cookbooks/equilibrium_grain_size/plugins/equilibrium_grain_size.h

```diff
@@ -0,0 +1,497 @@
+                                                        const Point<dim> &position) const;
+
+        double density (const double temperature,
+                        const double pressure,
+                        const std::vector<double> &compositional_fields,
+                        const Point<dim> &position) const;
+
+        double compressibility (const double temperature,
+                                const double pressure,
+                                const std::vector<double> &compositional_fields,
+                                const Point<dim> &position) const;
+
+        double thermal_expansivity (const double temperature,
+                                    const double pressure,
+                                    const std::vector<double> &compositional_fields,
+                                    const Point<dim> &position) const;
+
+        /**
+         * Returns the p-wave velocity as calculated by HeFESTo.
+         */
+        double seismic_Vp (const double      temperature,
+                           const double      pressure,
+                           const std::vector<double> &compositional_fields,
+                           const Point<dim> &position) const;
+
+        /**
+         * Returns the s-wave velocity as calculated by HeFESTo.
+         */
+        double seismic_Vs (const double      temperature,
+                           const double      pressure,
+                           const std::vector<double> &compositional_fields,
+                           const Point<dim> &position) const;
+
+        /**
+         * Function that defines the phase transition interface
+         * (0 above, 1 below the phase transition).This is done
+         * individually for each transition and summed up in the end.
+         */
+        double
+        phase_function (const Point<dim> &position,
+                        const double temperature,
+                        const double pressure,
+                        const unsigned int phase) const;
+
+        /**
+         * Function that returns the phase for a given
+         * position, temperature, pressure and compositional
+         * field index.
+         */
+        unsigned int
+        get_phase_index (const Point<dim> &position,
+                         const double temperature,
+                         const double pressure) const;
+
+
+        /**
+         * list of depth, width and Clapeyron slopes for the different phase
+         * transitions and in which phase they occur
+         */
+        std::vector<double> transition_depths;
+        std::vector<double> transition_temperatures;
+        std::vector<double> transition_slopes;
+        std::vector<double> transition_widths;
+
+
+        /**
+         * The following variables are properties of the material files
+         * we read in.
+         */
+        std::string datadirectory;
```

Comment:
maybe call it `data_directory` to be consistent with our usual spelling for variable names